packages/components/src/card/index.tsx

```diff
@@ -30,53 +30,55 @@ type ElementProps< P, T extends TagName > = P &
 
 export type Props< T extends TagName > = ElementProps< OwnProps, T >;
 
-class Card< T extends TagName = 'div' > extends PureComponent< Props< T > > {
-	render(): JSX.Element {
-		const {
-			children,
-			className,
-			compact,
-			displayAsLink,
-			highlight,
-			tagName = 'div',
-			href,
-			target,
-			...props
-		} = this.props;
+const Card = < T extends TagName = 'div' >(
+	{
+		children,
+		className,
+		compact,
+		displayAsLink,
+		highlight,
+		tagName = 'div',
+		href,
+		target,
+		...props
+	}: Props< T >,
+	forwardedRef: Ref< any > // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
The type of the forwarded ref should be an instance of `T` or instance of `'a'` if `href` prop is present?

Is there a React type for "instance of `T`"?

Reply:
Yes, ideally if it was a simple polymorphic component we would simply pass T there and all would be fine... however because passing `href` overrides `tagName` in this component (to me this is a terrible design choice that overcomplicates things unnecessarily) we cannot do that. I can try to experiment with some conditional types but I think using `Ref<any>` is fine here.

Reply:
I agree that `Ref<any>` is fine (although, what about `Ref<unknown>` -- is it better?). And I don't have any ideas right now how to improve the types much.

If `tagName` could be only a native HTML tag, then the ref would always point to an `Element`. But it can be any React component instance. And if a functional component uses `useImperativeHandle`, then the instance interface is completely arbitrary.